### PR TITLE
updated test workflow

### PR DESCRIPTION
**Pull request by Spyro000**
- disable "begin" btn if test haven`t any tasks;
- hide "begin" btn if test already begun;
- add checkbutton to each tasks;
- add "cancel" and "finish" btn`s;

-add x32 vagrant script to package.json
-add start-backend to package.json

### CODE REVIEW

File: app/profile/test/test.controller.js [hunk 1]
```diff
@@ -28,33 +31,61 @@
             testResolve.$promise.then((res) => {
                 if (res.success) {
                     vm.test = res.data;
+                    setStartButton();
```
Comment (oleh-polishchuk) [inline app/profile/test/test.controller.js]: `vm.test.content = res.data;`
`vm.test.setup();`

File: app/profile/test/test.controller.js [hunk 2]
```diff
@@ -28,33 +31,61 @@
...
+                }
+            });
+        }
+
+        function setStartButton() {
+            if (vm.test.taskList.length != 0) {
+                vm.startBtnDisabled = false;
```
Comment (oleh-polishchuk) [inline app/profile/test/test.controller.js]: `vm.test.containTasks = Boolean(vm.test.taskList.length)`

File: app/profile/test/test.controller.js [hunk 3]
```diff
@@ -28,33 +31,61 @@
...
                 $location.path('/profile/tests');
             });
         }
 
+        function stopTest() {
+            userTestsResolve.$promise.then((res) => {
+                let userTest = res.data.find(val => val.test.id == vm.test.id);
+                if (userTest) {
+                    ProfileTestService.stopTest({ 'id': userTest.id }).$promise.then(() => {
+                        $location.path('/profile/tests');
```
Comment (oleh-polishchuk) [inline app/profile/test/test.controller.js]: Just reload page.
Also add $log('Test started successfully.'); and notification with the same message.

File: app/profile/test/test.controller.js [hunk 4]
```diff
@@ -28,33 +31,61 @@
...
-                    {id: 10, name: 'Amber Gates', platoon: 'Platoon 10', date: '12.07.2016'},
-                    {id: 11, name: 'Jared Head', platoon: 'Platoon 11', date: '14.04.2017'},
-                    {id: 12, name: 'Isaiah Cabrera', platoon: 'Platoon 12', date: '10.02.2018'}
+                    { id: 6, name: 'Blythe Stokes', platoon: 'Platoon 6', date: '22.08.2016' },
+                    { id: 7, name: 'Mechelle Gibbs', platoon: 'Platoon 7', date: '21.04.2017' },
+                    { id: 8, name: 'Ruth Douglas', platoon: 'Platoon 8', date: '06.01.2018' },
+                    { id: 9, name: 'Erin Morton', platoon: 'Platoon 9', date: '05.03.2017' },
+                    { id: 10, name: 'Amber Gates', platoon: 'Platoon 10', date: '12.07.2016' },
+                    { id: 11, name: 'Jared Head', platoon: 'Platoon 11', date: '14.04.2017' },
+                    { id: 12, name: 'Isaiah Cabrera', platoon: 'Platoon 12', date: '10.02.2018' }
```
Comment (oleh-polishchuk) [inline app/profile/test/test.controller.js]: Please, remove temp info

File: app/profile/test/test.controller.js [hunk 5]
```diff
@@ -6,15 +6,18 @@
         .module('app')
         .controller('ProfileTestController', ProfileTestController);
 
-    function ProfileTestController(UserService, Constants, AppModalService, testResolve, $location, ProfileTestService) {
+    function ProfileTestController(UserService, Constants, AppModalService, testResolve, $location, ProfileTestService, userTestsResolve, $log) {
         const vm = this;
 
         vm.test = {};
+        vm.startBtnDisabled = true;
+        vm.showStopBtn = false;
```
Comment (Spyro000) [inline app/profile/test/test.controller.js]: done

File: app/profile/test/test.controller.js (hunk 1 above)
Comment (Spyro000) [inline app/profile/test/test.controller.js]: done

File: app/profile/test/test.controller.js (hunk 2 above)
Comment (Spyro000) [inline app/profile/test/test.controller.js]: done

File: app/profile/test/test.controller.js [hunk 6]
```diff
@@ -28,33 +31,61 @@
...
+                        vm.showStopBtn = true;
+                    }
                 }
             });
         }
 
         function startTest() {
-            ProfileTestService.startTest({"testId": vm.test.id}).$promise.then(() => {
+            ProfileTestService.startTest({ "testId": vm.test.id }).$promise.then(() => {
                 $location.path('/profile/tests');
```
Comment (Spyro000) [inline app/profile/test/test.controller.js]: done

File: app/profile/test/test.controller.js (hunk 3 above)
Comment (Spyro000) [inline app/profile/test/test.controller.js]: done

File: app/profile/test/test.controller.js (hunk 4 above)
Comment (Spyro000) [inline app/profile/test/test.controller.js]: done

File: app/profile/test/test.service.js [hunk 7]
```diff
@@ -10,6 +10,8 @@
 
         this.getTest = ProfileTestFactory.get;
         this.startTest = UserTestsFactory.save;
+        this.getUserTests = UserTestsFactory.get;
+        this.stopTest = UserTestsFactory.delete;
```
Comment (Spyro000) [inline app/profile/test/test.service.js]: done

**Pull request merged**